common/djangoapps/third_party_auth/models.py

```diff
@@ -881,11 +918,17 @@ def get_config(self):
         conf['x509cert'] = ''
         conf['url'] = sso_url
 
-        # Add SAMLConfiguration appropriate for this IdP
-        conf['saml_sp_configuration'] = (
-            self.saml_configuration or
-            SAMLConfiguration.current(self.site.id, 'default')
-        )
+        # Add SAMLConfiguration appropriate for this IdP (updated logic)
+        saml_config = self.get_current_saml_configuration()
+        if not saml_config:
+            log.error(
+                'No SAMLConfiguration found for provider "%s" on site %s. '
+                'Create a SAMLConfiguration or check site configuration.',
+                self.name, self.site_id
+            )
+            raise AuthNotConfigured(provider_name=self.name)
```

Comment:
Do we need to have this work any differently than before? I'd prefer to minimize changes and not cause new bugs where we aren't sure about the changes.